### PR TITLE
Add component discovery.dockerswarm to flow

### LINKED ISSUE #2264: Flow: discovery.dockerswarm component

### PR DESCRIPTION
**Pull request by wildum**
#### Which issue(s) this PR fixes

Fixes #2264 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (rfratto) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:19]: For usage, it feels like we shouldn't provide example values, otherwise users may not understand that the role may sometimes not be `services` and that the host may be different (especially on Windows).

```suggestion
  host = "DOCKER_DAEMON_HOST"
  role = "SWARM_ROLE"
```

**Review by rfratto — APPROVED** [review]
LGTM pending review from Clayton for the docs

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:107]: Was it intentional to split services and role into two in-line code elements?

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:111]:
```suggestion
- `__meta_dockerswarm_service_id`: the ID of the service.
```

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:109]: Future thoughts for a section like this. We are inconsistent with initial caps on the description after the colon. Some topics have inital caps and others don't.  No need to fix this time around... Just something to consider for future improvements in consistency. I would lean towards an initial caps, but either way works.

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:129]: Another split inline code?

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:135]:
```suggestion
- `__meta_dockerswarm_task_container_id`: the container ID of the task.
```

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:140]:
```suggestion
- `__meta_dockerswarm_service_id`: the ID of the service.
```

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/discovery.dockerswarm.md:165]: Another split in-line code?

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (wildum) [reply docs/sources/flow/reference/components/discovery.dockerswarm.md:107]: Yeah I did the split because `role` refers to the attribute and `services` is one of the value that `role` can take. But it does look a bit odd. Should I merge both -> `services role`? Or should we change the wording?

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (wildum) [reply docs/sources/flow/reference/components/discovery.dockerswarm.md:109]: true, we recently added a lot of new components but it should slow down very soon. It will be the opportunity to do some cleanup and to improve the consistency

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/discovery.dockerswarm.md:107]: Ah ok, now I see what you were trying to indicate here.

I think I'd just remove the inline-code from around `role`. The sections are specific about the specific roles, `services`, `tasks`, and `nodes`.

So something like this:

```suggestion
The `services` role discovers all [Swarm services](https://docs.docker.com/engine/swarm/key-concepts/#services-and-tasks) and exposes their ports as targets. For each published port of a service, a single target is generated. If a service has no published ports, a target per service is created using the `port` attribute defined in the arguments.
```

changing it from this:
image

to this:
image

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/discovery.dockerswarm.md:129]:
```suggestion
The `tasks` role discovers all [Swarm tasks](https://docs.docker.com/engine/swarm/key-concepts/#services-and-tasks) and exposes their ports as targets. For each published port of a task, a single target is generated. If a task has no published ports, a target per task is created using the `port` attribute defined in the arguments.
```

File: docs/sources/flow/reference/components/discovery.dockerswarm.md
Comment (clayton-cornell) [reply docs/sources/flow/reference/components/discovery.dockerswarm.md:165]:
```suggestion
The `nodes` role is used to discover [Swarm nodes](https://docs.docker.com/engine/swarm/key-concepts/#nodes).
```

**Review by clayton-cornell — APPROVED** [review]

**Pull request merged**
